test/qa/test_settings.py

```diff
@@ -119,3 +120,251 @@ def test_technology_set_options(tech, proto, obfuscated):
+def test_set_defaults_when_logged_in_1st_set(tech, proto, obfuscated):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
+
+    daemon.restart() # Temporary solution to avoid Firewall staying enabled in settings - LVPN-4121
+
+    sh.nordvpn.set.firewall("off")
+    sh.nordvpn.set.routing("off")
+    sh.nordvpn.set.dns("1.1.1.1")
+    sh.nordvpn.set.analytics("off")
+    sh.nordvpn.set.ipv6("on")
+    sh.nordvpn.set.notify("on")
+
+    assert not settings.get_is_firewall_enabled()
+    assert not settings.get_is_routing_enabled()
+    assert not settings.get_is_dns_disabled()
+    assert not settings.get_are_analytics_enabled()
+    assert settings.get_is_ipv6_enabled()
+    assert settings.get_is_notify_enabled()
+
+    if obfuscated == "on":
+        assert settings.get_is_obfuscated()
+    else:
+        assert not settings.get_is_obfuscated()
+
+    assert MSG_SET_DEFAULTS in sh.nordvpn.set.defaults()
+
+    assert settings.get_default_application_settings()
+
+
+@pytest.mark.parametrize(("tech", "proto", "obfuscated"), lib.STANDARD_TECHNOLOGIES) # Only using standard technologies here because of "LVPN-4601 - Enabling Auto-connect disables Obfuscation"
+def test_set_defaults_when_logged_in_2nd_set(tech, proto, obfuscated):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
+
+    sh.nordvpn.set.fwmark("0xe2f2")
+    sh.nordvpn.set.meshnet("on")
+    sh.nordvpn.set.killswitch("on")
+    sh.nordvpn.set.tpl("on")
+    sh.nordvpn.set.autoconnect("on")
+    sh.nordvpn.set("lan-discovery", "on")
+
+    assert "0xe1f1" not in sh.nordvpn.settings()
+    assert settings.is_meshnet_on()
+    assert daemon.is_killswitch_on()
+    assert settings.get_is_tpl_enabled()
+    assert settings.get_is_autoconnect_enabled()
+    assert settings.get_is_lan_discovery_enabled()
+    
+    if tech == "openvpn":
+        assert not settings.get_is_obfuscated()
+
+    assert MSG_SET_DEFAULTS in sh.nordvpn.set.defaults()
+
+    assert settings.get_default_application_settings()
+
+    login.login_as("default")
+    sh.nordvpn.set.meshnet("on")
+    meshnet.remove_all_peers()
+
+
+@pytest.mark.parametrize(("tech", "proto", "obfuscated"), lib.TECHNOLOGIES)
+def test_set_defaults_when_logged_out_1st_set(tech, proto, obfuscated):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
+
+    sh.nordvpn.set.fwmark("0xe2f2")
+    sh.nordvpn.set.killswitch("on")
+    sh.nordvpn.set.notify("on")
+    sh.nordvpn.set.meshnet("on")
+    sh.nordvpn.set.ipv6("on")
+    sh.nordvpn.set.dns("1.1.1.1")
+
+    assert "0xe1f1" not in sh.nordvpn.settings()
+    assert daemon.is_killswitch_on()
+    assert settings.get_is_notify_enabled()
+    assert settings.is_meshnet_on()
+    assert settings.get_is_ipv6_enabled()
+    assert not settings.get_is_dns_disabled()
+    
+    if obfuscated == "on":
+        assert settings.get_is_obfuscated()
+    else:
+        assert not settings.get_is_obfuscated()
+
+    sh.nordvpn.logout("--persist-token")
```

Comment:
this might be an app bug, but if the user logs out, shouldn't the app switch to the default settings automatically?

Reply:
Good question, settings are not reset to defaults upon logout, we do not have such a bug registered though, so not sure if this is the expected behavior or not. Do you think we should treat this as a bug?

Reply:
Discussed with PO, this will not be treated as a bug